### PR TITLE
fix: `submit-consumer-misbehaviour` command in the provider module CLI

### PR DESCRIPTION
**Pull request by sainoe**
## Description

Fix the unmarshalling of consumer misbehaviour from `json` files in the the `NewSubmitConsumerMisbehaviourCmd` of the provider module client. 
First, the misbehaviour `json` file wasn't properly opened, and second, the required IBC misbehaviour type wasn't registered to the provider's codec `InterfaceRegistry`.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

* [ ] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] Added `!` to the type prefix if state-machine breaking change (i.e., requires coordinated upgrade)
* [ ] Confirmed this PR does not introduce changes requiring state migrations, OR migration code has been added to consumer and/or provider modules
* [ ] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#pr-targeting))
* [ ] Provided a link to the relevant issue or specification
* [ ] Followed the guidelines for [building SDK modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/00-intro.md)
* [ ] Included the necessary unit and integration [tests](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#testing)
* [ ] Added a changelog entry to `CHANGELOG.md`
* [ ] Included comments for [documenting Go code](https://blog.golang.org/godoc)
* [ ] Updated the relevant documentation or specification
* [ ] Reviewed "Files changed" and left comments if necessary 
* [ ] Confirmed all CI checks have passed
* [ ] If this PR is library API breaking, bump the go.mod version string of the repo, and follow through on a new major release

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

* [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] confirmed `!` in the type prefix if API or client breaking change
* [ ] confirmed this PR does not introduce changes requiring state migrations, OR confirmed migration code has been added to consumer and/or provider modules
* [ ] confirmed all author checklist items have been addressed
* [ ] reviewed state machine logic
* [ ] reviewed API design and naming
* [ ] reviewed documentation is accurate
* [ ] reviewed tests and test coverage

### CODE REVIEW

File: x/ccv/provider/client/cli/tx.go [hunk 1]
```diff
@@ -101,11 +102,18 @@ Example:
 			txf = txf.WithTxConfig(clientCtx.TxConfig).WithAccountRetriever(clientCtx.AccountRetriever)
 
 			submitter := clientCtx.GetFromAddress()
-			var misbehaviour ibctmtypes.Misbehaviour
-			if err := clientCtx.Codec.UnmarshalInterfaceJSON([]byte(args[1]), &misbehaviour); err != nil {
+			misbRaw, err := os.ReadFile(args[0])
```
Comment (insumity) [inline x/ccv/provider/client/cli/tx.go:105]: Maybe rename `misbRaw` to `misbehaviourJson` similarly to what is being done in double-voting evidence with:
```
evidenceJson, err := os.ReadFile(args[0])
```
?

**Comment by insumity** [comment]
Would it be possible to add what exactly was the bug and how this change fixes it in the `Description` of this PR if it doesn't close a specific issue?

**Review by MSalopek — COMMENTED** [review]
Need some tips for testing this.

Otherwise LGTM

**Comment by insumity** [comment]
Looks good to me.

I was wondering:
1) Does this mean that currently a relayer won't be able to successfully submit a consumer light client attack? If so, do we need to backport this PR?
2) Is there a way to test this? If yes, it would make sense to have a test in this well for possible regressions in the future and as a way to verify that indeed the issue is fixed.

**Comment by sainoe** [comment]
> Does this mean that currently a relayer won't be able to successfully submit a consumer light client attack? If so, do we need to backport this PR?

No,  only submitting light client attacks through the CLI is currently unavailable. The Hermes Relayer uses the provider GRPC endpoint. It will be backported to v3.3.x.

> Is there a way to test this? If yes, it would make sense to have a test in this well for possible regressions in the future and as a way to verify that indeed the issue is fixed.

Agree. At the moment none of the CLI commands are tested, but it's something that we should probably do. @MSalopek is there any plan for that matter?

**Review by insumity — APPROVED** [review]

**Comment by MSalopek** [comment]
> > Does this mean that currently a relayer won't be able to successfully submit a consumer light client attack? If so, do we need to backport this PR?
> 
> No, only submitting light client attacks through the CLI is currently unavailable. The Hermes Relayer uses the provider GRPC endpoint. It will be backported to v3.3.x.
> 
> > Is there a way to test this? If yes, it would make sense to have a test in this well for possible regressions in the future and as a way to verify that indeed the issue is fixed.
> 
> Agree. At the moment none of the CLI commands are tested, but it's something that we should probably do. @MSalopek is there any plan for that matter?

No plans for a standalone CLI testing suite. You need a lot of state to test CLIs so e2e tests are not that dissimilar to your idea - e2e tests send CLI cmds anyway

**Review by MSalopek — APPROVED** [review]
LGTM

**Pull request merged**